config_parse.go

```diff
@@ -41,7 +43,8 @@ func ReadProxyConfig(path string) (c ProxyConfig, err error) {
 }
 
 // ReadConfig unmarshals the config file and slurps in it's data.
-func ReadConfig(path string) (c Config, err error) {
+func ReadConfig(path string) (*Config, error) {
```

Comment:
Why did this become a `*Config`? I think for configs especially, it's fairly peaceful to have them be returned as plain values...